lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -296,7 +296,7 @@ def test_put_bad_keys(self, client, server_config, attach_dataset):
         )
         assert response.status_code == HTTPStatus.BAD_REQUEST
         assert response.json == {
-            "message": "Unrecognized JSON keys ['what', 'xyzzy'] given for parameter metadata; allowed namespaces are ['dashboard', 'user']"
+            "message": "Unrecognized JSON keys ['what', 'xyzzy'] given for parameter metadata; allowed namespaces are ['dashboard', 'dataset.name', 'server.deletion', 'user']"
```

Comment:
Only those specific key paths are allowed by the keyword check, which is sensitive to key namespace. (That is, we allow `user.xyzzy` because the `user` namespace is allowed for `PUT`; but only `dataset.name` and `server.deletion` are listed for those namespaces.)